### PR TITLE
Add cmdstanpy

### PR DESCRIPTION
**Pull request by maresb**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by maresb** [comment]
@conda-forge/help-python ready for review!

File: recipes/cmdstanpy/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/stan-dev/cmdstanpy/archive/v{{ version }}.tar.gz
+  sha256: 3eede296981e2e6ad674699efdb358ce7da3e4f2add628ee4b377ef893405168
+
+build:
+  number: 0
+  skip: True  # [py<37]
```
Comment (mariusvniekerk) [inline recipes/cmdstanpy/meta.yaml:15]:
```suggestion
  noarch: python
```

File: recipes/cmdstanpy/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (mariusvniekerk) [inline recipes/cmdstanpy/meta.yaml:21]:
```suggestion
    - python >=3.7,<4.0
```

File: recipes/cmdstanpy/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  run:
+    - python
```
Comment (mariusvniekerk) [inline recipes/cmdstanpy/meta.yaml:23]:
```suggestion
    - python >=3.7,<4.0
```

**Review by mariusvniekerk — COMMENTED** [review]
Can this be made noarch with the following changes?

**Comment by maresb** [comment]
Fails on Windows with:
```
Packaging cmdstanpy
INFO conda_build.build:bundle_conda(1514): Packaging cmdstanpy-0.9.75-pyhd8ed1ab_0
Packaging cmdstanpy-0.9.75-pyhd8ed1ab_0
number of files: 20
WARNING: failed to get_linkages, codefile_type('C:\Miniconda\conda-bld\cmdstanpy_1618851500110\_h_env\Scripts\install_cmdstan.exe')=EXEfile
WARNING: failed to get_linkages, codefile_type('C:/Miniconda/conda-bld/cmdstanpy_1618851500110/_h_env\Scripts\install_cxx_toolchain.exe')=EXEfile
   INFO: sysroot: 'C:/Windows/' files: '['winhlp32.exe', 'win.ini', 'twain_32/wiatwain.ds', 'twain_32.dll']'
   INFO (cmdstanpy): Interpreted package 'cmdstanpy' is interpreted by 'python'
[noarch_python] Noarch package contains binary script: install_cmdstan.exe
##[error]Cmd.exe exited with code '1'.
```

**Comment by maresb** [comment]
Hi @mariusvniekerk, could you please take another look? It fails due to a console script named `install_cmdstan`. Are console scripts simply incompatible with noarch due to Windows?

**Comment by maresb** [comment]
@mariusvniekerk sorry about my ignorance here! After declaring the entrypoints it succeeds as noarch. (Sometimes it helps to RTFM. :wink:) Thanks for the advice!

**Pull request merged**